Gems/AtomLyIntegration/EMotionFXAtom/Code/Tools/EMStudio/AnimViewportWidget.cpp

```diff
@@ -0,0 +1,103 @@
+        setSizePolicy(qSize);
+        setAutoFillBackground(false);
+        setStyleSheet(QString::fromUtf8(""));
+
+        m_renderer = AZStd::make_unique<AnimViewportRenderer>(GetViewportContext());
+
+        SetupCameras();
+        SetupCameraController();
+    }
+
+    void AnimViewportWidget::SetupCameras()
+    {
+        m_pivotRotateCamera = AZStd::make_shared<AzFramework::RotateCameraInput>(EMStudio::ViewportUtil::BuildRotateCameraInputId());
+
+        const auto translateCameraInputChannelIds = EMStudio::ViewportUtil::BuildTranslateCameraInputChannelIds();
+        m_pivotTranslateCamera = AZStd::make_shared<AzFramework::TranslateCameraInput>(
+            translateCameraInputChannelIds, AzFramework::LookTranslation, AzFramework::TranslatePivot);
+        m_pivotTranslateCamera.get()->m_translateSpeedFn = []
+        {
+            return 3.0f;
+        };
+
+        m_pivotDollyScrollCamera = AZStd::make_shared<AzFramework::PivotDollyScrollCameraInput>();
+    }
+
+    void AnimViewportWidget::SetupCameraController()
+    {
+        auto controller = AZStd::make_shared<AtomToolsFramework::ModularViewportCameraController>();
+        controller->SetCameraViewportContextBuilderCallback(
+            [viewportId =
+                 GetViewportContext()->GetId()](AZStd::unique_ptr<AtomToolsFramework::ModularCameraViewportContext>& cameraViewportContext)
+            {
+                cameraViewportContext = AZStd::make_unique<AtomToolsFramework::ModularCameraViewportContextImpl>(viewportId);
+            });
+
+        controller->SetCameraPriorityBuilderCallback(
+            [](AtomToolsFramework::CameraControllerPriorityFn& cameraControllerPriorityFn)
+            {
+                cameraControllerPriorityFn = AtomToolsFramework::DefaultCameraControllerPriority;
+            });
+
+        controller->SetCameraPropsBuilderCallback(
+            [](AzFramework::CameraProps& cameraProps)
+            {
+                cameraProps.m_rotateSmoothnessFn = []
+                {
+                    return EMStudio::ViewportUtil::CameraRotateSmoothness();
+                };
+
+                cameraProps.m_translateSmoothnessFn = []
+                {
+                    return EMStudio::ViewportUtil::CameraTranslateSmoothness();
+                };
+
+                cameraProps.m_rotateSmoothingEnabledFn = []
+                {
+                    return EMStudio::ViewportUtil::CameraRotateSmoothingEnabled();
+                };
+
+                cameraProps.m_translateSmoothingEnabledFn = []
+                {
+                    return EMStudio::ViewportUtil::CameraTranslateSmoothingEnabled();
+                };
+            });
+
+        controller->SetCameraListBuilderCallback(
+            [this](AzFramework::Cameras& cameras)
+            {
+                cameras.AddCamera(m_pivotRotateCamera);
+                cameras.AddCamera(m_pivotTranslateCamera);
+                cameras.AddCamera(m_pivotDollyScrollCamera);
```

Comment:
suggestion: We could also add `Pan` too (middle mouse button) - or I can help with this when the code is in 👍